AssetRipper.Library/Exporters/Textures/YamlSpriteExporter.cs

```diff
@@ -0,0 +1,81 @@
+using AssetRipper.Core.Classes.Misc;
+using AssetRipper.Core.Interfaces;
+using AssetRipper.Core.Parser.Files.SerializedFiles;
+using AssetRipper.Core.Project.Collections;
+using AssetRipper.Core.Project.Exporters;
+using AssetRipper.SourceGenerated.Classes.ClassID_213;
+using AssetRipper.SourceGenerated.Classes.ClassID_687078895;
+using AssetRipper.SourceGenerated.Subclasses.SecondarySpriteTexture;
+using AssetRipper.SourceGenerated.Subclasses.SpriteAtlasData;
+using AssetRipper.SourceGenerated.Subclasses.SpriteRenderData;
+
+namespace AssetRipper.Library.Exporters.Textures
+{
+	/// <summary>
+	/// Exports sprites as YAML assets.
+	/// </summary>
+	public sealed class YamlSpriteExporter : YamlExporterBase
+	{
+		public override bool IsHandle(IUnityObjectBase asset) => asset is ISprite or ISpriteAtlas;
+
+		public override IExportCollection CreateCollection(VirtualSerializedFile virtualFile, IUnityObjectBase asset)
+		{
+			if (asset is ISpriteAtlas)
+			{
+				return new EmptyExportCollection();
+			}
+			
+			// SpriteAtlas is a new feature since Unity 2017.
+			// For older versions of Unity, SpriteAtlas doesn't exist,
+			// and the correct metadata of Sprite is stored in the m_RD field.
+			// Otherwise, if a SpriteAtlas reference is serialized into this sprite,
+			// we must recover the m_RD field of the sprite from the SpriteAtlas.
+			ISprite sprite = (ISprite)asset;
+			if (sprite.Has_SpriteAtlas_C213() && sprite.Has_RenderDataKey_C213() && sprite.SpriteAtlas_C213.FindAsset(sprite.SerializedFile) is { } atlas)
+			{
+				ISpriteAtlasData spriteData = atlas.RenderDataMap_C687078895[sprite.RenderDataKey_C213];
+				ISpriteRenderData m_RD = sprite.RD_C213;
+				IUnityObjectBase? texture = spriteData.Texture.FindAsset(atlas.SerializedFile);
+				if (texture is not null)
+				{
+					m_RD.Texture.CopyValues(sprite.SerializedFile.CreatePPtr(texture));
+				}
+				if (m_RD.Has_AlphaTexture())
+				{
+					m_RD.AlphaTexture.CopyValues(spriteData.AlphaTexture);
+				}
```

Comment:
`AlphaTexture` has to be handled similarly to `Texture`.

Reply:
fixed this, together with secondary textures